### PR TITLE
ISLANDORA-1087: fixes WSOD

### PR DESCRIPTION
**Pull request by DiegoPino**
https://jira.duraspace.org/browse/ISLANDORA-1087

Fixes orphan DOMNodes associated to deleted Form elements when
processing the XML document on submit in Multistep workflows. 

Since all  objects used by XMLForms are stored inside $form_state and serialised 
by drupal to keep multistep form states, when the XML document is processed,
and after delete actions are run, we end with DOMNodes stored in the internal registry
that do not have parents, and also DOMAttrib associated to them. When
serialisation of this particular object happens every DOMNode is
converted to an XPath. On unserialise, those XPaths are not valid
agains the XML document and results in an silent exception when the
unserialization process is nested, destroying $form_state and by this
making form steps impossible.

### CODE REVIEW

**Comment by DiegoPino** [comment]
This solution was the one with less new code. I did some tests going back and forth as many times i could using many types of XML Forms, with and without templates. 
But there could be a particular edge case when a very deep nested attribute associated to a particular complex XML could fail. I could not make this fail after the fix, but since this is very tricky i would love second opinions on this.  I can't review my own code. @whikloj ? @ruebot ? @nigelgbanks any thoughts on this since you are the Father? :flushed: Any particular complex XML form someone is willing to contribute to make this stuff fail?

File: api/XMLFormProcessor.inc [hunk 1]
```diff
@@ -198,6 +201,44 @@ class XMLFormProcessor {
...
+   *   The element to check against.
+   */
+  protected function cleanupNodeRegistry(FormElement $element) {
+    // Checks if the form elements associated with a DOMNodes where
+    // removed. If any found, verifies if the associated DOMNodes are
+    // attributes of an orphan node or orphan ones, removing those from
+    // nodeRegistry.
+    // This is needed because the serialization and unserialization
+    // process of NodeRegistry with orphan Domnodes results in invalid
+    // XPath's that silently fail when unserializing the supper
```
Comment (ruebot) [inline api/XMLFormProcessor.inc]: super :smile:

**Comment by DiegoPino** [comment]
@ruebot ready!
